integration-tests/runtime-tests/test/tests/mosaic/mosaicTests.ts

```diff
@@ -0,0 +1,419 @@
+import { expect } from 'chai';
+import {ApiPromise, Keyring} from "@polkadot/api";
+import testConfiguration from './test_configuration.json';
+import { sendAndWaitForSuccess } from '@composable/utils/polkadotjs';
+import {KeyringPair} from "@polkadot/keyring/types";
+import { mintAssetsToWallet } from '@composable/utils/mintingHelper';
+import { send } from 'process';
+import {
+    CommonMosaicRemoteAssetId,
+    OrmlTokensAccountData,
+    PalletMosaicDecayBudgetPenaltyDecayer,
+    PalletMosaicNetworkInfo
+} from "@composable/types/interfaces";
+import {start} from "repl";
+import {AccountId} from "@polkadot/types/interfaces/runtime";
+import { doesNotMatch } from 'assert';
+
+/**
+ * Mosaic Pallet Tests
+ *  Checked functionalities are as follows;
+ *  1. setRelayer
+ *  2. rotateRelayer
+ *  3. setNetwork
+ *  4. setBudget
+ *  5. transferTo
+ *  6. acceptTransfer
+ *  7. claimStaleTo
+ *  8. timelockedMint
+ *  9. setTimelockDuration
+ * 10. rescindTimelockedMint
+ * 11. claimTo
+ * 12. updateAssetMapping
+ *
+ * This suite consists of happy path tests. Additionally, we started implementing suites for later references such as regression, smoke etc.
+ *
+ */
+describe('tx.mosaic Tests', function () {
+    // Check if group of tests are enabled.
+    if (!testConfiguration.enabledTests.query.enabled)
+        return;
+    let sudoKey: KeyringPair,
+        startRelayerWallet: KeyringPair,
+        newRelayerWallet: KeyringPair,
+        userWallet: KeyringPair,
+        remoteAssetId: CommonMosaicRemoteAssetId;
+    let transferAmount: number,
+        assetId: number,
+        networkId: number;
+    let pNetworkId;
+    let ethAddress;
+
+    describe('tx.mosaic Success Tests', function() {
+        this.timeout(4*60*1000);
+        if (!testConfiguration.enabledTests.query.account__success.enabled)
+            return;
+        before('Initialize variables',async function() {
+            this.timeout(4 * 60 * 1000);
+            sudoKey = walletAlice;
+            startRelayerWallet = walletEve.derive("/tests/mosaicPallets/wallet1");
+            newRelayerWallet = walletAlice.derive("/tests/mosaicPallets/wallet2");
+            userWallet = walletFerdie.derive("/tests/mosaicPallets/wallet3");
+            assetId = 4;
+            transferAmount = 100000000000;
+            networkId = 1;
+            pNetworkId = api.createType('u128', 1);
+            ethAddress = "0x";
+            remoteAssetId = api.createType('CommonMosaicRemoteAssetId', {
+                EthereumTokenAddress: api.createType('[u8;20]', "0x")
+            });
+        });
+        before('Mint available assets into wallets', async function(){
+            this.timeout(2*60*1000);
```

Comment:
```suggestion
            this.timeout(5*60*1000);
```
Please increase the timeout a bit. It seems to occasionally fail therefore.